### DISCUSSION TITLE
Deceptive site ahead

### DISCUSSION THREAD

**Discussion by maitrungduc1410** [nextauthjs/next-auth#6393] [Help] [answered]
### Environment

System:
    OS: macOS 11.3.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 480.87 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.18.0 - /usr/local/bin/node
    Yarn: 1.22.11 - /usr/local/bin/yarn
    npm: 8.19.2 - /usr/local/bin/npm
    Watchman: 2022.09.26.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 109.0.5414.87
    Firefox: 102.0
    Safari: 14.1
  npmPackages:
    next: 13.0.1 => 13.0.1 
    next-auth: ^4.15.0 => 4.15.0 
    react: 18.2.0 => 18.2.0 

### Reproduction URL

https://app.txtecommerce.asia/api/auth/error

### Describe the issue

We're facing this issue in production, where Chrome/Safari/Firefox show `Deceptive site ahead` on all endpoints related to NextAuth

I'm using NextAuth + Keycloak, everything is very basic, I just follow same tutorial of NextAuth: https://next-auth.js.org/providers/keycloak

The issue came and caused our user scare because Red screen is being shown.

I check Google Search Console and this is list of all affected endpoints:

<img width="775" alt="Screenshot 2023-01-14 at 11 14 12 PM" src="https://user-images.githubusercontent.com/16630193/212482569-1446ee5f-3da1-4c9a-b8fd-50ff65f95cca.png">

### How to reproduce

Simply access: https://app.txtecommerce.asia

### Expected behavior

It shouldn't show the warning page

**Comment by fgarciamacias** [comment]
Hi @maitrungduc1410, I'm facing the same issue how do you solve it, also I'm working with nextauth + keycloak, I can't find a solution, I hope you can help me. Thanks

**Comment by vincentchu** [comment]
I ran into this same issue running using `next-auth` + Vercel.

At the end of the day, the fix for me was quite simple. I simply changed the `id` attribute on the `GoogleProvider` [configuration](https://github.com/nextauthjs/next-auth/blob/ead715219a5d7a6e882a6ba27fa56b03954d062d/src/providers/google.js#L3) to `"g"` (default is `"google"`).

This has the effect of changing the signin login to `/api/auth/signin/g` from `/api/auth/signin/google`. My suspicion is that Chrome is being very aggressive with paths that contain `google` as a token with other things that suggest authentication or logging in though I'm not quite sure and haven't been able to quite reproduce this on other sites/domains, even with the same path. 

This is what my `[...nextauth].ts` looks like now: 

```ts
import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";

export const authOptions = {
  // Configure one or more authentication providers
  providers: [
    GoogleProvider({
      id: "g", // CHANGE THIS
      clientId: process.env.GOOGLE_CLIENT_ID || "",
      clientSecret: process.env.GOOGLE_CLIENT_SECRET || "",
    }),
  ],
};

export default NextAuth(authOptions);
```

![Screenshot 2023-07-01 at 6 12 31 AM](https://github.com/nextauthjs/next-auth/assets/27597/b8c9cba9-2398-4c13-ac96-77efe96765aa)

**Comment by maitrungduc1410** [comment, marked as answer]
I figured out that there's a bug for an endpoint in my app.

In my authOptions I fetch some user info stored in Redis.

And the bug causes Redis overload -> /api/auth/session is hanged, never returns -> error page is showed. And when Google indexes my app, it runs through the buggy endpoint and it sees error page -> marks my website as "Deceptive"

After resolving the issue, its fine now.

Hope it can help someone

**Reply by jrusk** [reply]
I had a similar problem. I was trying to fix an issue with `EmailProvider` and Microsoft Outlook Safelinks. In particular I used code like this comment in [issue 6436](https://github.com/nextauthjs/next-auth/issues/6436#issuecomment-1504263781). The Google bots didn't seem to like this and so they marked all my auth API routes as malicious. I used the [Google Search Console](https://search.google.com/search-console/security-issues) and then removed my code detecting bots.  After that I selected the option in the Google Search Console to review my site and the next day I got an email saying all was clear and my site was back to normal.
